### DISCUSSION TITLE
`sitemap.xml` is dynamic in NextJS 15 (used to be static)

### DISCUSSION THREAD

**Discussion by MartinXPN** [vercel/next.js#72013] [Help] [answered]
### Summary

I generate the sitemap with `sitemap.ts`.
When accessing the `/sitemap.xml` route, the sitemap is generated on every request.
It used to be generated once and served statically.
Am I missing something? How can I make sure the `sitemap` is generated at build time once?

### Additional information

This seems to be a new thing with NextJS 15. Used to be static with NextJS 14.

**Comment by icyJoseph** [comment]
How is the sitemap setup, is the route segment dynamic? It'd be great to have a reproduction repository, or to see how it is being used.

**Reply by MartinXPN** [reply]
It's a `sitemap.ts` in the `src/app` directory.

**Reply by icyJoseph** [reply]
Errrr 🤔 I'll try to setup a reproduction repository. Does your build output also show it as a dynamic path? It uses a lambda icon for those.

**Reply by MartinXPN** [reply]
Yeah, I'm getting this after the build step: `ƒ /sitemap.xml`

**Reply by icyJoseph** [reply]
- Alright, have you searched the issues section to see if this has been reported?

**Reply by MartinXPN** [reply]
Yeah, I tried to find similar issues but didn't notice anyone reporting this for the newest version.

**Reply by MartinXPN** [reply]
Oh, I think I found the issue. I was calling `revalidateTag` from the `sitemap.ts` to fetch fresh data (as the `unstable_cache` was messing with `Firestore.Timestamp` object and I needed to use the timestamp to populate the `lastModified` field).
After removing the `revalidateTag` and `unstable_cache`, the sitemap is generated statically.

**Reply by icyJoseph** [reply, marked as answer]
Yeah, that'd make sense, the sitemap is opt-in to dynamic behavior by doing that.
